shared/hooks/useRelativeTime.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from 'react';
+
+import { formatDistanceToNowStrict } from 'date-fns';
+import { enUS, ru } from 'date-fns/locale';
+import { useRouter } from 'next/router';
+
+const localeMap = {
+  en: enUS,
+  ru,
+};
+
+type Locale = 'en' | 'ru';
+
+type Replacements = {
+  [key: string]: string;
+};
+
+const replacementsMap: Record<Locale, Replacements> = {
+  en: {
+    minute: 'min',
+    minutes: 'min',
+    month: 'mo',
+    months: 'mo',
+    second: 'sec',
+    seconds: 'sec',
+  },
+  ru: {
+    года: 'г',
+    дней: 'д',
+    дня: 'д',
+    лет: 'г',
+    месяц: 'мес',
+    месяцев: 'мес',
+    минут: 'мин',
+    минуты: 'мин',
+    недели: 'нед',
+    неделя: 'нед',
+    секунд: 'сек',
+    секунда: 'сек',
+    секунды: 'сек',
+    часа: 'ч',
+    часов: 'ч',
+  },
+};
+
+const getShortRelativeTime = (date: Date, locale: Locale): string => {
+  const distance = formatDistanceToNowStrict(date, { locale: localeMap[locale] });
+  const replacements = replacementsMap[locale];
+
+  const shortDistance = distance
+    .split(' ')
+    .map(word => replacements[word] || word)
+    .join(' ');
+
+  return locale === 'ru' ? `${shortDistance} назад` : `${shortDistance} ago`;
+};
+
+const useRelativeTime = (date: string): string => {
+  const { locale } = useRouter();
+  const effectiveLocale = locale === 'ru' || locale === 'en' ? locale : 'en';
+  const [relativeTime, setRelativeTime] = useState<string>(
+    getShortRelativeTime(new Date(date), effectiveLocale)
+  );
+
+  useEffect(() => {
```

Comment:
`useEffect` and `useState` are obviously unnecessary here. If the locale is changed, the component will be re-rendered anyway, which will provoke `getShortRelativeTime` call. Therefore, it is enough to simply return the call of this function. And the timer inside the effect at all almost gave me a heart attack =D.

_By the way_, if you remove this effect with a timer, then at the same time will disappear annoying error that looms in the console every second.